api/src/main/java/openconsensus/OpenConsensus.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019, OpenConsensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package openconsensus;
+
+import java.util.concurrent.atomic.AtomicReference;
+import openconsensus.trace.NoopTrace;
+import openconsensus.trace.Trace;
+import openconsensus.trace.Tracer;
+
+/**
+ * Registry provides a global access to observability objects. An implementation is responsible for
+ * setting the Registry.
+ */
+public class OpenConsensus {
+
+  private static final OpenConsensus INSTANCE = new OpenConsensus();
+
+  private final AtomicReference<Tracer> tracer = new AtomicReference<>(Trace.getTracer());
```

Comment:
With the current design of the `OpenConsensus` class:
1. It is possible to call `new OpenConsensus()` from anywhere.
1. What is the purpose of the complexity behind `getInstance()` on line 60, in combination with the static `INSTANCE` variable on line 30? Both `getInstance()` on line 60 and the static `INSTANCE` variable on line 30 can be removed. Thereafter, declare `tracer` on line 32 as `static`, and reference it directly from all static methods in the `OpenConsensus` class.

or...

The only reason I can think of to have an openly instantiable `OpenConsensus` class is for inheritance (i.e. if the design of this class intends to allow users to be able to extend the structural and functional behavior of  `OpenConsensus` class with `class MyOpenConsensus extends OpenConsensus` semantics. However, I do not believe this is a good pattern to follow. I believe `OpenConsensus` should have a private constructor, thus also making the class final.

Reply:
Related to 1. the private constructor was removed when moving things around. Thanks for the catch!

2. `getInstance` will be more complicated once we introduce service loader I have removed it for now to keep it clean.

We do not want people extending this call.

Reply:
Also, part of (2) in my post was regarding `INSTANCE` and `tracer`. To recap, `INSTANCE` is not necessary, as `tracer` can be made static. This is what @carlosalberto is saying also.

Additionally, polymorphism does not apply to `getInstance`, as it is a static method (i.e. nobody could extend it even if it was kept around).